### PR TITLE
Amd64/misc fixes

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Misc fixes for amd64
It has been confirmed by julenuri that this PR does also fix JIRA issue: [CORE-18155](https://jira.reactos.org/browse/CORE-18155)

### CODE REVIEW

File: sdk/lib/rtl/slist.c [hunk 1]
```diff
@@ -104,9 +104,79 @@ RtlInterlockedPushListSList(
...
+        /* ListHead and List must be in the same region */
+        ASSERT(((ULONG64)SListHead & 0xFFFFF80000000000ull) ==
+               ((ULONG64)List & 0xFFFFF80000000000ull));
+
+        /* Read the header */
+        OldSListHead = *SListHead;
+
+        do
+        {
+            /* Construct the first entry from the header and the  */
```
Comment (HBelusca) [inline sdk/lib/rtl/slist.c:157]: "and the ..." ?

File: sdk/lib/rtl/slist.c [hunk 2]
```diff
@@ -104,9 +104,79 @@ RtlInterlockedPushListSList(
...
+    if (RtlpUse16ByteSLists)
+    {
+         BOOLEAN exchanged;
+
+        do
+        {
+            /* Capture the current SListHead */
+            OldSListHead = *SListHead;
+
+
```
Comment (HBelusca) [inline sdk/lib/rtl/slist.c:122]:
```suggestion
```

File: sdk/lib/rtl/amd64/unwind.c [hunk 3]
```diff
@@ -1053,29 +1053,37 @@ RtlpCaptureNonVolatileContextPointers(
...
+        }
+        else
+        {
+            Context.Rip = *(PULONG64)Context.Rsp;
+            Context.Rsp += 8;
+        }
 
-        /* Continue until we reached the target frame or user mode */
-    } while (EstablisherFrame < TargetFrame);
+        /* Continue until we reached user mode */
```
Comment (HBelusca) [inline sdk/lib/rtl/amd64/unwind.c:1102]:
```suggestion
        /* Continue until we reach user mode */
```

File: hal/halx86/amd64/x86bios.c [hunk 4]
```diff
@@ -449,36 +443,42 @@ x86BiosCall(
     EmulatorContext.Flags.AlwaysSet = 1;
     EmulatorContext.Flags.If = 1;
 
+    /* Set up the INT stub */
+    FlatIp = StackBase - 4;
+    InstructionPointer = x86BiosMemoryMapping + FlatIp;
+    InstructionPointer[0] = 0xCD; // INT instruction
+    InstructionPointer[1] = (UCHAR)InterruptNumber;
+    InstructionPointer[2] = 0xD6; // undefined, used as return indicator
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply hal/halx86/amd64/x86bios.c:451]: Why isn't it possible to use the "BOP-code" 0xC4C4 ?

File: dll/win32/oleaut32/msvc.S [hunk 5]
```diff
@@ -12,10 +12,62 @@
...
+// call_method(
+//     void *func, // rcx
+//     int nb_args, // edx
+//     const DWORD_PTR *args );
 PUBLIC call_method
-call_method:
-    int 2ch
+FUNC call_method
+    push rbp
+    .PUSHREG rbp
```
Comment (HBelusca) [inline dll/win32/oleaut32/msvc.S:25]: So the `.PUSHREG` MASM thing doesn't generate any executable code? (thus the `push` opcode above is needed as well?)

File: hal/halx86/amd64/x86bios.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (tkreuzer) [reply hal/halx86/amd64/x86bios.c]: I'm using the address.

File: dll/win32/oleaut32/msvc.S (hunk 5 above)
Comment (tkreuzer) [reply dll/win32/oleaut32/msvc.S:25]: https://letmegooglethat.com/?q=.PUSHREG

File: modules/rostests/apitests/ntdll/NtContinue.c [hunk 6]
```diff
@@ -167,28 +228,38 @@ START_TEST(NtContinue)
...
+        {
+            Registers[i] = randULONG64();
+        }
+
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
+        continueContext.Dr0 = randULONG64() & 0xFFFF;
```
Comment (tkreuzer) [inline modules/rostests/apitests/ntdll/NtContinue.c:252]: Copypasta

File: modules/rostests/apitests/ntdll/amd64/NtContinue.S [hunk 7]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (tkreuzer) [inline modules/rostests/apitests/ntdll/amd64/NtContinue.S:1]: Add header

File: dll/win32/oleaut32/msvc.S (hunk 5 above)
Comment (HBelusca) [reply dll/win32/oleaut32/msvc.S:25]: That's precisely because I saw the MSDN "documentation" about it that I asked about that.

**Comment by julenuri** [comment]
As commented in the ticket [CORE-18155](https://jira.reactos.org/browse/CORE-18155), this fixes the changing resolution. 

This also fixes the ticket [CORE-18156](https://jira.reactos.org/browse/CORE-18156), allows the CD change.

I consider OK to merge it into master (if devs consider it nice). It doesn't cause any conflict into compilers as the CRT does.

**Review by JoachimHenze — APPROVED** [review]
I do give my consent to the PR now, although the stuff it deals with is totally over my head.
My consent is based on the positive testers feedback here just.

**Comment by IniKiwi** [comment]
and?

**Pull request merged**